src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ca/AssessmentService.java

```diff
@@ -928,4 +929,33 @@ private void checkNegNumber(Integer value) {
       throw new ValidationException(format(ERR_MSG_FMT_INVALID_CRITERION_INTEGER, value));
     }
   }
+
+  /**
+   *
+   * @param toolId
+   * @param dimensionId
+   * @param lotId
+   * @return
+   */
+  public String getSupplierDimensionData(final Integer toolId, final Integer dimensionId,
+      Integer lotId) {
+
+    // Explicitly validate toolId so we can throw a 404 (otherwise empty array returned)
+    var assessmentTool = retryableTendersDBDelegate.findAssessmentToolById(toolId).orElseThrow(
+        () -> new ResourceNotFoundException(format(ERR_MSG_FMT_TOOL_NOT_FOUND, toolId)));
+
+    var dimension = retryableTendersDBDelegate.findDimensionById(dimensionId).orElseThrow(
+        () -> new ResourceNotFoundException(format(ERR_MSG_FMT_DIMENSION_NOT_FOUND, dimensionId)));
+
+    Set<Integer> supplierSubmissions ;
+
+    if (lotId != null && lotId > 0) {
+      supplierSubmissions = retryableTendersDBDelegate.findAssessmentTaxonByToolIdAndDimensionIdAndLotId(assessmentTool.getId(),
+          dimension.getId(),lotId);
+    } else {
+      supplierSubmissions = retryableTendersDBDelegate.findAssessmentTaxonByToolIdAndDimensionId(assessmentTool.getId(),
+          dimension.getId());
+    }
+    return supplierSubmissions.stream().map(i -> i.toString()).collect(Collectors.joining(", "));
```

Comment:
Have you seen the output of this - do we know it is in the right format - this looks like it would just concatenate the rows together with a comma (rather than format it correctly as a CSV)?